pyuntl/untldoc.py

```diff
@@ -675,3 +676,43 @@ def find_untl_errors(untl_dict, **kwargs):
         'error_dict': error_dict,
     }
     return found_data
+
+
+def untl_dict_to_tuple(untl_dict):
+    """Convert untl_dict values to list of lists of tuples."""
+    untl_tuple = {}
+    for elem in untl_dict:
+        for i, v in enumerate(untl_dict[elem]):
+            # We are trying to get a consistent ordering of values
+            # so reordering doesn't count as a change.
+            if isinstance(untl_dict[elem][i]['content'], dict):
+                untl_dict[elem][i]['content'] = [v for v in untl_dict[elem][i]['content'].items()]
+                untl_dict[elem][i]['content'].sort()
+            untl_dict[elem][i] = [v for v in untl_dict[elem][i].items()]
+        untl_tuple[elem] = untl_dict[elem]
+    return untl_tuple
+
+
+def generate_hash(input):
+    """Return an md5 hash of an object by first converting it to bytes."""
+    return hashlib.md5(repr(input).encode()).hexdigest()
+
+
+def untl_to_hash_dict(untl_elements, meaningfulMeta=True):
+    """Convert untl element to a dictionary of hashed values."""
+    untl_dict = untlpy2dict(untl_elements)
+    if meaningfulMeta:
+        remove_elements = []
+        for element in untl_dict['meta']:
+            if (
+                element['qualifier'] == 'metadataModificationDate'
+                or element['qualifier'] == 'metadataModifier'
+            ):
+                remove_elements.append(element)
+        for element in remove_elements:
+            untl_dict['meta'].remove(element)
+    untl_tuple = untl_dict_to_tuple(untl_dict)
+    hash_dict = {}
+    for key, value in untl_tuple.items():
+        hash_dict[key] = generate_hash(value)
+    return hash_dict
```

Comment:
The above 4 lines could go down to 1 with a dictionary comprehension:
```
    return {k: generate_hash(v) for k, v in untl_tuple.items()}
```
Though, what you have is fine--I am just seeing comprehensions everywhere today.